### PR TITLE
dnf: update `use_backend` documentation

### PR DESCRIPTION
**Pull request by flowerysong**
##### SUMMARY

Provide clarity on what each valid option (including the previously undocumented `dnf`) means.

##### ISSUE TYPE

- Docs Pull Request

### CODE REVIEW

File: lib/ansible/modules/dnf.py [hunk 1]
```diff
@@ -19,9 +19,15 @@
 options:
   use_backend:
     description:
-      - By default, this module will select the backend based on the C(ansible_pkg_mgr) fact.
+      - Backend module to use.
```
Comment (Akasurde) [inline lib/ansible/modules/dnf.py:22]: Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

Comment (flowerysong) [reply lib/ansible/modules/dnf.py:22]: There's no actual change here, though. It's just documentation, which isn't normally called out in the changelog.

Comment (Akasurde) [reply lib/ansible/modules/dnf.py:22]: ok.

**Review by mkrizek — APPROVED** [review]
Thanks!

File: lib/ansible/modules/dnf.py [hunk 2]
```diff
@@ -19,9 +19,15 @@
...
     default: "auto"
-    choices: [ auto, yum, yum4, dnf4, dnf5 ]
+    choices:
+        auto: Automatically select the backend based on the C(ansible_facts.pkg_mgr) fact.
+        yum: Alias for V(auto)
```
Comment (bcoca) [inline lib/ansible/modules/dnf.py]: explain that yum is kept for backwards compatibility and that for 'real' yum you should go to older ansible versions.

Comment (bcoca) [reply lib/ansible/modules/dnf.py]: can just have a 'see notes' and explain this in those

**Review by bcoca — APPROVED** [review]

**Pull request merged**